pandas/core/dtypes/cast.py

```diff
@@ -19,15 +20,28 @@
     pandas_dtype)
 from .dtypes import DatetimeTZDtype, ExtensionDtype, PeriodDtype
 from .generic import (
-    ABCDatetimeArray, ABCDatetimeIndex, ABCPeriodArray, ABCPeriodIndex,
-    ABCSeries)
+    ABCDatetimeArray, ABCDatetimeIndex, ABCIndexClass, ABCPeriodArray,
+    ABCPeriodIndex, ABCSeries)
 from .inference import is_list_like
 from .missing import isna, notna
 
 _int8_max = np.iinfo(np.int8).max
 _int16_max = np.iinfo(np.int16).max
 _int32_max = np.iinfo(np.int32).max
 _int64_max = np.iinfo(np.int64).max
+_int64_min = np.iinfo(np.int64).min
+_uint64_max = np.iinfo(np.uint64).max
+_float32_max = np.finfo(np.float32).max
+
+
+def _is_iNaT(x):
+    if not is_scalar(x):
+        return False
+    with warnings.catch_warnings():
+        # bug in numpy warnings for timedelta, see numpy/numpy#10095
```

Comment:
we already have several routines for this
don’t think we need another

Reply:
I don't think there is one, but happy to be corrected. If you look at the numpy-bug I'm linking, you'll see that this just happens in certain timedelta cases; these would not fail per se, but cause warnings which then fail the build.